src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -0,0 +1,353 @@
+              source.getInputStream(partitionStartByteRange, partitionEndByteRange);
+          partitionInputStream = new SequenceInputStream(partitionInputStream, overlapStream);
+          parser =
+              new CSVParser(
+                  new BufferedReader(new InputStreamReader(partitionInputStream)),
+                  CSVFormat.newFormat(delimiter).withQuote(quote).withEscape(escape));
+          iterator = parser.iterator();
+          byteOverlap *= 2;
+        } else {
+          discardedRecord = true;
+        }
+      } else if (!discardedRecord) {
+        for (int column = 0; column < schema.numColumns(); ++column) {
+          String cell = record.get(column);
+          try {
+            switch (schema.getColumnType(column)) {
+              case BOOLEAN_TYPE:
+                if (Floats.tryParse(cell) != null) {
+                  buffer.putBoolean(column, Floats.tryParse(cell) != 0);
+                } else if (BooleanUtils.toBoolean(cell)) {
+                  buffer.putBoolean(column, Boolean.parseBoolean(cell));
+                }
+                break;
+              case DOUBLE_TYPE:
+                buffer.putDouble(column, Double.parseDouble(cell));
+                break;
+              case FLOAT_TYPE:
+                buffer.putFloat(column, Float.parseFloat(cell));
+                break;
+              case INT_TYPE:
+                buffer.putInt(column, Integer.parseInt(cell));
+                break;
+              case LONG_TYPE:
+                buffer.putLong(column, Long.parseLong(cell));
+                break;
+              case STRING_TYPE:
+                buffer.putString(column, cell);
+                break;
+              case DATETIME_TYPE:
+                buffer.putDateTime(column, DateTimeUtils.parse(cell));
+                break;
+            }
+          } catch (final IllegalArgumentException e) {
+            throw new DbException(
+                "Error parsing column "
+                    + column
+                    + " of row "
+                    + lineNumber
+                    + ", expected type: "
+                    + schema.getColumnType(column)
+                    + ", scanned value: "
+                    + cell,
+                e);
+          }
+          if (onLastRow) {
+            parser.close();
+          }
+        }
+      }
+    }
+
+    LOGGER.debug("Scanned {} input lines", lineNumber - lineNumberBegin);
+
+    return buffer.popAny();
+  }
+
+  @Override
+  public void cleanup() throws IOException {
+    parser = null;
+    while (buffer.numTuples() > 0) {
+      buffer.popAny();
+    }
+  }
+
+  @Override
+  protected Schema generateSchema() {
+    return schema;
+  }
+
+  @Override
+  protected void init(final ImmutableMap<String, Object> execEnvVars) throws DbException {
+    buffer = new TupleBatchBuffer(getSchema());
+    try {
+      partitionInputStream = source.getInputStream(partitionStartByteRange, partitionEndByteRange);
```

Comment:
Weren't we going to pad the first `InputStream` we fetch by `BYTE_OVERLAP_PARALLEL_INGEST`, since the last record beginning in the partition's range will normally be truncated?